src/lib.rs

```diff
@@ -59,8 +61,9 @@ pub fn decompressed_len(compressed: &[u8]) -> Result<usize, InvalidInput> {
 
 /// Compress a buffer using snappy.
 pub fn compress(input: &[u8]) -> Vec<u8> {
-	let mut buf = Vec::new();
-	compress_into(input, &mut buf);
+	let mut buf = Vec::with_capacity(0);
```

Comment:
What's the diff between `Vec::with_capacity(0)` and `Vec::new()`?

Reply:
We will resize the array to the expected size inside `compress_into`, my idea was to use `with_capacity(0)` to avoid any eager allocation.

Reply:
I believe `new()` does not allocate either, so it should be the same.